frontend/src/XbddRouter.tsx

```diff
@@ -26,12 +27,31 @@ interface ReportIdentifier {
 
 const ReportPage: FC<Props> = ({ user }) => {
   const dispatch = useDispatch();
+  const location = useLocation();
+
   const { productParam, versionParam, buildParam } = useParams<ReportIdentifier>();
   const { product, version, build } = getDecodedIdentifier(productParam, versionParam, buildParam);
-  dispatch(updateReportIdentifier(product, version, build));
-  dispatch(fetchIndexes());
-  dispatch(fetchTagsMetadata());
-  dispatch(fetchProductList());
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    const hiddenStatuses = searchParams.getAll('hidden');
+    const selectedStatuses = {
+      [Passed]: !hiddenStatuses.includes(Passed),
+      [Failed]: !hiddenStatuses.includes(Failed),
+      [Skipped]: !hiddenStatuses.includes(Skipped),
+      [Undefined]: !hiddenStatuses.includes(Undefined),
+    };
+    dispatch(saveStatusFilter({ selectedStatuses }));
+  }, [dispatch, location.search]);
+
+  useEffect(() => {
+    dispatch(updateReportIdentifier(product, version, build));
+
+    dispatch(fetchIndexes());
```

Comment:
Better pattern for this stuff (potentially in a different service) would be something like the below - so it doesn't mix rest and redux together:
fetchIndexes().then(indexes => dispatch(retrievedIndexes(indexes));